src/main/java/com/hubspot/jinjava/util/DeferredValueUtils.java

```diff
@@ -0,0 +1,208 @@
+package com.hubspot.jinjava.util;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import com.hubspot.jinjava.interpret.Context;
+import com.hubspot.jinjava.interpret.DeferredValue;
+import com.hubspot.jinjava.interpret.JinjavaInterpreter;
+import com.hubspot.jinjava.tree.ExpressionNode;
+import com.hubspot.jinjava.tree.Node;
+import com.hubspot.jinjava.tree.TagNode;
+import com.hubspot.jinjava.tree.TextNode;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.StringJoiner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class DeferredValueUtils {
+  private static final String TEMPLATE_TAG_REGEX = "(\\w+(?:\\.\\w+)*)";
```

Comment:
I don't think this will work with multivariable assignments like `set myvar1, myvar2, myvar3, myvar4 = ...` I'm not sure you'd want to though as that might make things very complex.

Reply:
Nice catch. I added this regex to be more specific for looking for sets as we don't have the existing context to compare matches to filter out noise. I think I will change this (and the other usages) to filter out any reserved tokens instead.